frontend/packages/text-editor/src/utils.ts

```diff
@@ -52,9 +52,9 @@ export const filterFunction = (
     return true;
   } else if (searchQuery.length < 1) {
     return true;
-  } else if (id?.includes(searchQuery)) {
+  } else if (id.toLowerCase()?.includes(searchQuery.toLowerCase())) {
```

Comment:
according to the type, both the id and value may be a string or undefined. We should check if the id or value is **not** undefined before we invoke the toLowerCase function. Then we avoid errors such as "Cannot read properties of undefined (reading 'toLowerCase')" :) 

```suggestion
  } else if (id?.toLowerCase()?.includes(searchQuery.toLowerCase())) {
```